spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/LoadBalancerAutoConfiguration.java

```diff
@@ -47,13 +48,22 @@
  * @author Will Tran
  * @author Gang Li
  * @author Olga Maciaszek-Sharma
+ * @author Andrii Bohutskyi
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnClass(RestTemplate.class)
 @ConditionalOnBean(LoadBalancerClient.class)
-@EnableConfigurationProperties(LoadBalancerProperties.class)
+@EnableConfigurationProperties({ LoadBalancerProperties.class, LoadBalancerServiceProperties.class })
 public class LoadBalancerAutoConfiguration {
 
+	@Bean
+	@ConditionalOnMissingBean
+	public LoadBalancerPropertiesFactory loadBalancerPropertiesFactory(LoadBalancerProperties properties,
+			LoadBalancerServiceProperties servicesProperties,
+			@Value("${spring.cloud.loadbalancer.service.configuration.enabled:false}") boolean isServiceProperties) {
```

Comment:
Let's also add this property to `additional-spring-configuration-metadata.json`.